lib/kube/proxy/cluster_details.go

```diff
@@ -78,13 +83,14 @@ func (k *kubeDetails) Close() {
 }
 
 // getKubeClusterCredentials generates kube credentials for dynamic clusters.
-// TODO(tigrato): add support for aws logins via token
 func getKubeClusterCredentials(ctx context.Context, cloudClients cloud.Clients, cluster types.KubeCluster, log *logrus.Entry, checker ImpersonationPermissionsChecker) (kubeCreds, error) {
 	switch {
 	case cluster.IsKubeconfig():
 		return getStaticCredentialsFromKubeconfig(ctx, cluster, log, checker)
 	case cluster.IsAzure():
 		return getAzureCredentials(ctx, cloudClients, cluster, log, checker)
+	case cluster.IsAWS():
```

Comment:
The `getKubeClusterCredentials` seems to be called also for static resource. 
Can we ensure that flow where kube access is done by teleport-kube-agent  inside k8 cluster works ? 
Where default `kube-agnet-config` configuration looks like below: 
```
    kubernetes_service:
      enabled: true
      kube_cluster_name: gcp
      labels:
        env: dev
```